src/context/PreviewEvents.js

```diff
@@ -0,0 +1,113 @@
+import React, {
+  useCallback,
+  useRef,
+  useState,
+  useContext,
+  createContext,
+} from 'react';
+import throttle from 'lodash.throttle';
+
+import { addLoggingEvents } from '../lib/domEvents';
+
+const PreviewEventsContext = createContext();
+const PreviewEventsActionsContext = createContext();
+
+export function PreviewEventsProvider({ children }) {
+  const buffer = useRef([]);
+  const previewRef = useRef();
+  const sortDirection = useRef('asc');
+  const [appendMode, setAppendMode] = useState('bottom');
+  const [eventCount, setEventCount] = useState(0);
+  const [eventListeners, setEventListeners] = useState([]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Maybe this line is not needed anymore because no `react-hooks/exhaustive-deps` is defined in `eslintrc.js`.